### PR TITLE
raise ValueError when logical date is earlier than start_date

### LINKED ISSUE #26865: logical_date<start_date causes Null DAGRun
### Apache Airflow version

2.4.1

### What happened

When you trigger a DAG that has a logical date earlier then the DAGs start date. You get a successful DAG Runs, but none of the tasks run:
image

### What you think should happen instead

All of the tasks in the DAGRun should execute with a logical date of the one supplied

### How to reproduce

- Have DAG with start_date of A
- Trigger DAG w/ Config, set logical_date before A 
- look at DAG Run state and notice none of the Tasks have run

### Operating System

Debian

### Versions of Apache Airflow Providers

_No response_

### Deployment

Astronomer

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Soonmok**
if logical_date if earlier than start_date and logical_date is set by config, the trigger logic doesn't catch the Error.

Fixes: https://github.com/apache/airflow/issues/26865

### CODE REVIEW

File: airflow/api/common/trigger_dag.py [hunk 1]
```diff
@@ -51,7 +51,11 @@ def _trigger_dag(
     if dag is None or dag_id not in dag_bag.dags:
         raise DagNotFound(f"Dag id {dag_id} not found")
 
-    execution_date = execution_date if execution_date else timezone.utcnow()
+    if execution_date is None:
+        if conf and "logical_date" in conf:
+            execution_date = conf["logical_date"]
```
Comment (uranusjr) [inline airflow/api/common/trigger_dag.py]: I don’t recall specifying `logical_date` in trigger config being a valid usage? Why is this added?

Comment (Soonmok) [reply airflow/api/common/trigger_dag.py]: In this [tutorial](https://airflow.apache.org/docs/apache-airflow/1.10.1/tutorial.html), default_args doesn't provide logical_date, but the issuer was assigning `execution_date` to default_args. 

I think we should check validation of using default_args.

Comment (Soonmok) [reply airflow/api/common/trigger_dag.py]: Maybe we can set whitelist.

Comment (Soonmok) [reply airflow/api/common/trigger_dag.py]: https://github.com/apache/airflow/blob/401fc57e8ba1dddb041e0d777bb0277a09f227db/airflow/models/dag.py#L295 
The comment describe, default_arg values can be conflict with operator's arguments, it will follow to operator's argument. 
Should `execution_date` argument follow this rule?

Comment (Taragolis) [reply airflow/api/common/trigger_dag.py]: `default_args` != `dag_run.conf`

1. https://airflow.apache.org/docs/apache-airflow/stable/dag-run.html#passing-parameters-when-triggering-dags
2. https://airflow.apache.org/docs/apache-airflow/stable/tutorial/fundamentals.html#tasks

Comment (Taragolis) [reply airflow/api/common/trigger_dag.py]: `dag_run.conf` use in Templates, and `default_args` parameters propagate to tasks arguments if it not set explicitly.

Comment (Soonmok) [reply airflow/api/common/trigger_dag.py]: Thank you for the explanation. 
Than, the issuer might have used `default_args` to parse `logical_date` (execution_date)

Comment (Soonmok) [reply airflow/api/common/trigger_dag.py]: I've change conf to default_args.

Comment (Taragolis) [reply airflow/api/common/trigger_dag.py]: And this also incorrect assumption.

`default_args` designed for reuse same parameters across Tasks within the single DAG, and it user-defined on code level. This arguments do not changed by their nature in runtime, except selected arguments:

1. `params` - [Params](https://airflow.apache.org/docs/apache-airflow/stable/concepts/params.html#params) could defined on DAG and Task level, so it resolved during DAG initialisation
2. `start_date` and `end_date` - https://airflow.apache.org/docs/apache-airflow/stable/faq.html#what-s-the-deal-with-start-date

That is all exceptions. 


`default_args` doesn't not contain `logical_date` / `execution_date` unless user define it, and this might happen if required propagate this arguments to [TriggerDagRunOperator](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/trigger_dagrun/index.html#airflow.operators.trigger_dagrun.TriggerDagRunOperator) or [ExternalTaskSensor](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/sensors/external_task/index.html#airflow.sensors.external_task.ExternalTaskSensor) without explicit define it during initialisation. But even in this case most probably it would be something like:

```python

with DAG(
    ..., 
    default_args={
        "execution_date": "{{ logical_date }}"
    },
) as dag:
    ...
```

Comment (Soonmok) [reply airflow/api/common/trigger_dag.py]: Okay. 
According to the [issue](https://github.com/apache/airflow/issues/26865), the issuer triggered the dag with config. so I assumed it was triggered with execution_date which is set by default_args.
but now I think there is another reason of this issue. I'll look it up more.
